### ISSUE TITLE
Is there a way to run object detection only in a specific area of the Image?

### ISSUE THREAD

**Issue by panca1** [AlexeyAB/darknet#6887]
Hi everybody, I have to run object detection in 4k images, but often on images there are many parts where I haven't to run object detections. (For example the sky) Is there a way to do it? 
I don't want to cut the image, because in output I want also the area that I haven't to consider when I have to do object detection.

**Comment by stephanecharette** [comment]
> Is there a way to do it?

...to do what exactly?  Your question is not clear.  Even if the image is the sky, just run it like you would normally.  Post an example of the command you want to run, and then perhaps show why or how that command is failing.

**Comment by panca1** [comment]
No, I don't have any command. My question is if there is a way to exclude/ignore a part of image from object detection? 
For example if I have an image with 60% of it is road, and 40% of it is sky. Can I ignore the sky from object detection and do it only on the road?

**Comment by stephanecharette** [comment]
Yes, either use the usual RoI method when passing in the cv::Mat (but that assumes you already know the coordinates of certain things...like the sky and the road) or through training you teach your network to either identify or ignore the sky, depending on what you want done with it.

**Comment by stephanecharette** [comment]
But I still don't exactly understand the question.  Let's say you are training the network to find cars and trucks.  Presumably the only cars and trucks are on the road and not in the sky.  So normally you'd train the network to find the things you want, and so if there happens to be a bunch of sky in the image, nothing would be annotated in it, and so it is ignored.

Or do you mean you are looking for things like birds, and you only want the birds on the ground versus the birds in the sky?

**Comment by panca1** [comment]
No. First option is the point. 

" Let's say you are training the network to find cars and trucks. Presumably the only cars and trucks are on the road and not in the sky. So normally you'd train the network to find the things you want, and so if there happens to be a bunch of sky in the image, nothing would be annotated in it, and so it is ignored." 

I want to do this

**Comment by stephanecharette** [comment]
> No. First option is the point.

Then you don't need to do anything.  The sky, trees, buildings, etc will automatically be ignored as long as your training images are representative of the final images on which you plan to run detection.

If you are just getting started with Darknet/YOLO, you may want to run through a simple tutorial:  https://www.ccoderun.ca/programming/2020-03-07_Darknet/

Also good to browse through if you are new to this:  https://www.ccoderun.ca/darkmark/ImageMarkup.html

**Comment by seung-gu** [comment]
> But I still don't exactly understand the question. Let's say you are training the network to find cars and trucks. Presumably the only cars and trucks are on the road and not in the sky. So normally you'd train the network to find the things you want, and so if there happens to be a bunch of sky in the image, nothing would be annotated in it, and so it is ignored.

Hi @stephanecharette I have a little question on that.
Doesn't Yolo use grid cells though the whole image to see if there is an object in each grid cell with anchor boxes? Then, I don't think the sky part is automatically ignored, since the grids still exist in the whole area. So, if ROI is set a half of image, then the grid cells of the other half will be completely ignored, and this results in a slightly faster speed. Maybe I'm wrong. Please let me know if I'm wrong.
